backend/src/main/java/atlantbh/restaurants/controllers/LoginController.java

```diff
@@ -0,0 +1,42 @@
+package atlantbh.restaurants.controllers;
+
+import atlantbh.restaurants.controllers.dto.LoginRequestDTO;
+import atlantbh.restaurants.controllers.dto.LoginResponseDTO;
+import atlantbh.restaurants.models.User;
+import atlantbh.restaurants.services.TokenService;
+import atlantbh.restaurants.services.UserService;
+import org.hibernate.service.spi.ServiceException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class LoginController extends BaseController<User, UserService> {
+
+    public @ResponseBody
+    ResponseEntity<LoginResponseDTO> login(@RequestBody LoginRequestDTO request) {
+        try {
+            User user = service.checkLoginData(request.getEmail(), request.getPassword());
+
+            if (user != null) {
```

Comment:
Controller should not know for null check. This code should be in `LoginService`. 
Controller needs to be something like ```return ResponseEntity.status(HttpStatus.OK).body(service.login(parameters));```

`service.login` should return LoginResponseDTO or throw service exception if something goes wrong.

Reply:
Fixed in 92c4f7264b274d7da369717e0ed1478a93d30ee4